src/oc_trace_context_headers.erl

```diff
@@ -0,0 +1,42 @@
+%%%-------------------------------------------------------------------------
+%% Copyright 2017, OpenCensus Authors
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% @doc Functions to support the http header format of the tracecontext-spec
+%% @end
+%%%-------------------------------------------------------------------------
+-module(oc_trace_context_headers).
+
+-export([encode/1,
+         decode/1]).
+
+-include("opencensus.hrl").
+
+-define(VERSION, "00").
+
+-spec encode(opencensus:trace_context()) -> iolist().
+encode(#trace_context{trace_id=TraceId,
+                      span_id=SpanId,
+                      enabled=Enabled}) ->
+    Options = case Enabled of true -> <<"01">>; _ -> <<"00">> end,
+    EncodedTraceId = io_lib:format("~32.16.0b", [TraceId]),
+    EncodedSpanId = io_lib:format("~16.16.0b", [SpanId]),
+    [?VERSION, "-", EncodedTraceId, "-", EncodedSpanId, "-", Options].
+
+-spec decode(binary()) -> opencensus:trace_context().
+decode(TraceContext) when is_list(TraceContext) ->
+    decode(list_to_binary(TraceContext));
+decode(<<?VERSION, "-", TraceId:32/binary, "-", SpanId:16/binary, "-", Enabled:2/binary, _Rest/binary>>) ->
+    #trace_context{trace_id=binary_to_integer(TraceId, 16),
```

Comment:
I think it would be better to return a tuple of `{ok, <trace>}` and `{error, invalid}` the specs point out that both a 0 as traceid or spanid are invalid and treating them as such seems to be the more proper implementation.

Reply:
That would also allow for not crashing on the library side for an invalid header and leaving that decision up to the caller (either case or pattern match on the result) and give a more helpful error i.e. `"badmatch {error, invalid}"` isntead of function clause not matching